### DISCUSSION TITLE
Warning for "always False" reference checking?

### DISCUSSION THREAD

**Discussion by Holt59** [microsoft/pyright#7835] [Q&A]
I just noticed that pyright does not warn for improper use of `is` between variables of different types, e.g.

```python
def f(x: dict[int, float], y: float):
    if y is x:  # <- no warning/error here
        print("foo?")
```

Is there a reason for that? Is there a way to enable such warning?

I am asking because I just spend some times because I misspelled `is` instead of `in`, so that would have been useful.

**Comment by erictraut** [comment]
No one has ever requested such a feature before. I don't recall seeing any such request in the mypy issue tracker either, so this must be a relatively uncommon situation. 

I'd consider adding such a feature if there's enough signal from pyright users that it would be useful. If you run across this thread and would like to see such a feature added, please add a "thumbs up" reaction.
